frontends/p4/p4-parse.ypp

```diff
@@ -813,6 +827,34 @@ actionRef
                                            $$ = new IR::ActionListElement(@2, $1, mce); }
     ;
 
+entry
+    : optAnnotations keysetExpression ':' actionInvocation
+                                         {
+                                           assert($4->is<IR::MethodCallExpression>());
+                                           if ($2->is<IR::ListExpression>())
+                                             $$ = new IR::Entry(@1+@4, $1,
+                                                                $2->to<IR::ListExpression>(), $4);
+                                           else {  // if not a tuple, make it a list of 1
+                                             auto le = new IR::Vector<IR::Expression>();
+                                             le->push_back($2);
+                                             $$ = new IR::Entry(@1+@4, $1,
+                                                                new IR::ListExpression(@2, le),
+                                                                $4);
+                                           }
+                                         }
+    ;
+
+actionInvocation
```

Comment:
actionBinding or actionConstruction is a more suitable name. The action is not being invoked.